src/diffusers/models/unet_2d_condition.py

```diff
@@ -123,20 +124,45 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embed_dim: int = None,
+        time_embedding_type: str = "positional", # fourier, positional
+        timestep_act_2: bool = False, 
+        class_labels_dim: Optional[int] = None,
+        conv_in_kernel: int = 3,
+        conv_out_kernel: int = 3,
+        norm_out: bool = True,
+        attn1_types: Tuple[bool, None]=(None,'self','self','self'),
```

Comment:
This is getting too complex I think. Let's see whether we can solve this differently (we probably should create a new attention layer here)

Reply:
yep! 

or maybe we don't need an attention layer at all - because it's just cross_attention/self_attention with a little bit of modification; I can directly use the 'CrossAttention' class inside the block